### PR TITLE
New: Discovery mode API to enable/disable discovery mode

### PR DESCRIPTION
**Pull request by ericrpowers**
#### Description
Discovery mode is going to get its own API. One will now be able to deploy and remove Discovery Mode by namespace (while specifying propagation or not) without the need of a recipe.

### CODE REVIEW

File: specs/designmode.spec [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# Model
+model:
+  rest_name: designmode
+  resource_name: designmodes
```
Comment (t00f) [inline specs/designmode.spec:4]: Let's just call it designmode without s

POST|DELETE /designmode

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "design-mode-api",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "a83b6f000ea12aaa8ad1a22efc2aa1df32c70444",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "b379d96f4a5107407d816e0a59c6b734fef7c5ae",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "design-mode-api",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "37a332afa7f847aa9fed6739bda9161cd6ed1281",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "0d74c28668697e79be11d36d6d3d1fe3ff431b11",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]
This one looks good to me, thanks

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "design-mode-api",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "4e786490fe768acde24a55785232c1ef0623069c",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "1cf45b6aeb6857b5a64a4644b66ef1ffa155af6a",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "0d74c28668697e79be11d36d6d3d1fe3ff431b11",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "design-mode-api",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "a2c6d7ea20c5d9e7f0ce0a27e51a88edf7493c9a",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "83cd342413c35a789e7469e7f66706efe1260f7f",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "61025c7f3664894939a7ad1340c49035036527f2",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "design-mode-api",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "c0a317f32f9c6f46af6b845bee7f5abe3cd03ba2",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "1ee305d73f91cd7d7fdefb6450f2c8c10f5516aa",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "a78a03d7d17e4aa8d5651562c2007b5854485694",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "design-mode-api",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "c9fac4ffd8eafbd00d83b82e0113478b50355fe6",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "194d63ad3489e7fddf5e7e23e8eed41fd013fa71",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "14e719628d0015de7ee378c63e247e43a2cb91d9",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "010fabb15c685e87d8bf31768edd55fb45123ef4",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "194d63ad3489e7fddf5e7e23e8eed41fd013fa71",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "194d63ad3489e7fddf5e7e23e8eed41fd013fa71",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "2f3af8eacf4bc0ec6e091f7d34f18ac5122395cb",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "194d63ad3489e7fddf5e7e23e8eed41fd013fa71",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "010fabb15c685e87d8bf31768edd55fb45123ef4",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "194d63ad3489e7fddf5e7e23e8eed41fd013fa71",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "4fd08649064b019c94403e2d47a65653258e0817",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "194d63ad3489e7fddf5e7e23e8eed41fd013fa71",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "2b756d4f07962a474d67b2a2c5f6708f5478ffb8",
    "pipeline": "master"
  }
]
```

**Review by t00f — DISMISSED** [review]
Thanks for changing the name 🙏

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "design-mode-api",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "8f78f79e88911a12f79b59023b14bca91e560eb2",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "194d63ad3489e7fddf5e7e23e8eed41fd013fa71",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "dbc9950e47db72c7805c2ee756edb2c9781a7a2e",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "194d63ad3489e7fddf5e7e23e8eed41fd013fa71",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "8cedc4481b474ad04e9852c7a07e8545c3c2e359",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "design-mode-api",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "ced267d7364d1b5985d44ef7303229b39245fa9b",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "7c030aa7705b341b3ec6e3cf14e91a82e00ad58d",
    "pipeline": "master"
  },
  {
    "project": "design-mode-api",
    "component": "apotests",
    "pr-id": "2966",
    "commit-sha": "8cedc4481b474ad04e9852c7a07e8545c3c2e359",
    "pipeline": "master"
  }
]
```

**Comment by ericrpowers** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "",
    "component": "gaia",
    "pr-id": "567",
    "commit-sha": "7c030aa7705b341b3ec6e3cf14e91a82e00ad58d",
    "pipeline": "master"
  },
  {
    "project": "",
    "component": "backend",
    "pr-id": "904",
    "commit-sha": "0c1a6e7697237514a4a6fdf9ad1b606b1a5096fe",
    "pipeline": "master"
  }
]
```

**Pull request merged**